### ISSUE TITLE
Separate ML models from the main `hidimstat` library

### ISSUE THREAD

**Issue by jpaillard** [mind-inria/hidimstat#18]
Due to the current implementation of BBI, the adaptation of ML model such as DnnLearner and ModifiedRF is part of the main library. 

I would argue to keep these modules outside of the main library as they are not statistical inference methods. Either by keeping them in repositories specific to publications or a separate folder dedicated to experimental code. 

Also related to issue #17 , suggesting that VIM method take a fitted model (compatible with sklearn API) rather than taking care of the model fit.

**Comment by achamma723** [comment]
> @achamma723 what was the motivation for introducing these models ?

Hello,
these are the models by default that mark the balanced performance of BBI. DNN_learner is implemented in sklearn-compatible format as the default inference learner and the Modifier (or Sampling RF) as the default importance learner

**Comment by bthirion** [comment]
Thx for the explanation. I understand that these models can be useful ---but they also have a cost.
I think that the best way forward is to have this in a submodule called learners`.

**Comment by tbng** [comment]
A bit nitpicking but I noticed that the API used in DNN learner scripts is quite different from older modules -- most of the objects inside the learners are Python classes and not functions.

**Comment by lionelkusch** [comment]
@achamma723 By looking at the code of BBI packages. I was wondering why you do not use the scikit-learn packages for doing the same thing based on:
- [parameterGrid](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.ParameterGrid.html) for the hyper-tuning of the estimator
- [MultiOutputClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.multioutput.MultiOutputClassifier.html) and [MultiOutputRegressor](https://scikit-learn.org/stable/modules/generated/sklearn.multioutput.MultiOutputRegressor.html) for the management of multiple target estimation
- [BaggingClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.BaggingClassifier.html) and [BaggingRegressor](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.BaggingRegressor.html) for the management of bagging estimators
- [ColumnTransformer](https://scikit-learn.org/stable/modules/generated/sklearn.compose.ColumnTransformer.html) with [OneHotEncoder](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html), [OrdinalEncoder](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OrdinalEncoder.html) and [StandardScaler](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.StandardScaler.html) for encoding and scale the data before the estimation in a [Pipeline](https://scikit-learn.org/stable/modules/generated/sklearn.pipeline.Pipeline.html)
- [TransformedTargetRegressor](https://scikit-learn.org/stable/modules/generated/sklearn.compose.TransformedTargetRegressor.html) for the scaling target output in specific cases
- [root_mean_squared_error](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.root_mean_squared_error.html) and [log_loss](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.log_loss.html) for computing the loss of the estimator

**Comment by bthirion** [comment]
I need some time to investigate whether this model is really useful.

**Comment by bthirion** [comment]
Closing as done.
